sympy/sets/tests/test_conditionset.py

```diff
@@ -30,52 +31,78 @@ def test_CondSet():
     # not depend on the dummy symbol): the result is `y > 5`.
     # In this case, ConditionSet is just acting like
     # Piecewise((Interval(1, 7), y > 5), (S.EmptySet, True)).
-    raises(TypeError, lambda: 6 in ConditionSet(x, y > 5, Interval(1, 7)))
+    raises(TypeError, lambda: 6 in ConditionSet(x, y > 5,
+        Interval(1, 7)))
 
-    assert isinstance(ConditionSet(x, x < 1, {x, y}).base_set, FiniteSet)
+    assert isinstance(ConditionSet(x, x < 1, {x, y}).base_set,
+        FiniteSet)
     raises(TypeError, lambda: ConditionSet(x, x + 1, {x, y}))
     raises(TypeError, lambda: ConditionSet(x, x, 1))
+    raises(BadSignatureError, lambda: ConditionSet((x, x), x))
 
     I = S.Integers
     C = ConditionSet
+    assert C(x, False, I) is S.EmptySet
+    assert C(x, True, I) is I
     assert C(x, x < 1, C(x, x < 2, I)
         ) == C(x, (x < 1) & (x < 2), I)
     assert C(y, y < 1, C(x, y < 2, I)
-        ) == C(x, (x < 1) & (y < 2), I)
+        ) == C(x, (x < 1) & (y < 2), I), C(y, y < 1, C(x, y < 2, I))
     assert C(y, y < 1, C(x, x < 2, I)
         ) == C(y, (y < 1) & (y < 2), I)
     assert C(y, y < 1, C(x, y < x, I)
         ) == C(x, (x < 1) & (y < x), I)
-    assert C(y, x < 1, C(x, y < x, I)
-        ) == C(L, (x < 1) & (y < L), I)
-    c = C(y, x < 1, C(x, L < y, I))
-    assert c == C(c.sym, (L < y) & (x < 1), I)
-    assert c.sym not in (x, y, L)
-    c = C(y, x < 1, C(x, y < x, FiniteSet(L)))
-    assert c == C(L, And(x < 1, y < L), FiniteSet(L))
+    assert unchanged(C, y, x < 1, C(x, y < x, I))
+    assert C((x, y), x + y > 0, C((w, z), w + z > 1, S.Integers)
+        ) == C((x, y), (x + y > 0) & (x + y > 1), S.Integers**2)
```

Comment:
Should this be over `Integers**2`?

This is either nonsensical or the empty set:
```
In [28]: C((w, z), w + z > 1, S.Integers)                                                                                                                     
Out[28]: {(w, z) | (w, z) ∊ ℤ ∧ (w + z > 1)}
```

Reply:
Entering a single set supplies the same set to all variables. This prints differently for me, however:
```
>>> c=ConditionSet((w, z), w + z > 1, S.Integers)
>>> c
ConditionSet((w, z), w + z > 1, ProductSet(Integers, Integers))
>>> pprint(c)
⎧                   2              ⎫
⎨(w, z) | (w, z) ∊ ℤ  ∧ (w + z > 1)⎬
⎩                                  ⎭
```